pims/cine.py

```diff
@@ -68,147 +79,204 @@ def _build_struct(dtype):
+    ('Res_bright', INT32),
+    ('Res_contrast', INT32),
+    ('Res_gamma', INT32),
+    ('Res_21', UINT32),
+    ('auto_exp_level', UINT32),
+    ('auto_exp_speed', UINT32),
     ('auto_exp_rect', RECT),
     ('wb_gain', '8f'),
-    ('rotate', INT),
+    ('rotate', INT32),
     ('wb_view', WBGAIN),
-    ('real_bpp', UINT),
-    ('conv_8_min', UINT),
-    ('conv_8_max', UINT),
-    ('filter_code', INT),
-    ('filter_param', INT),
+    ('real_bpp', UINT32),
+    ('Res_conv_8_min', UINT32),
+    ('Res_conv_8_max', UINT32),
+    ('filter_code', INT32),
+    ('filter_param', INT32),
     ('uf', IMFILTER),
-    ('black_cal_sver', UINT),
-    ('white_cal_sver', UINT),
-    ('gray_cal_sver', UINT),
+    ('black_cal_sver', UINT32),
+    ('white_cal_sver', UINT32),
+    ('gray_cal_sver', UINT32),
     ('b_stamp_time', BOOL),
-    ('sound_dest', UINT),
-    ('frp_steps', UINT),
-    ] + [('frp_img_nr%d' % i, INT) for i in range(16)] + \
-        [('frp_rate%d' % i, UINT) for i in range(16)] + \
-        [('frp_exp%d' % i, UINT) for i in range(16)] + [
-    ('mc_cnt', INT),
-    ] + [('mc_percent%d' % i, FLOAT) for i in range(64)] + [
-    ('ci_calib', UINT),
-    ('calib_width', UINT),
-    ('calib_height', UINT),
-    ('calib_rate', UINT),
-    ('calib_exp', UINT),
-    ('calib_edr', UINT),
-    ('calib_temp', UINT),
-    ] + [('header_serial%d' % i, UINT) for i in range(4)] + [
-    ('range_code', UINT),
-    ('range_size', UINT),
-    ('decimation', UINT),
-    ('master_serial', UINT),
-    ('sensor', UINT),
-    ('shutter_ns', UINT),
-    ('edr_shutter_ns', UINT),
-    ('frame_delay_ns', UINT),
-    ('im_pos_xacq', UINT),
-    ('im_pos_yacq', UINT),
-    ('im_width_acq', UINT),
-    ('im_height_acq', UINT),
-    ('description', '4096s')
+    ('sound_dest', UINT32),
+    ('frp_steps', UINT32),
+    ('frp_img_nr', '16'+INT32),
+    ('frp_rate', '16'+UINT32),
+    ('frp_exp', '16'+UINT32),
+    #] + [('frp_img_nr{:d}'.format(i), INT32) for i in range(16)] + \
+    #    [('frp_rate{:d}'.format(i), UINT32) for i in range(16)] + \
+    #    [('frp_exp{:d}'.format(i), UINT32) for i in range(16)] + [
+    ('mc_cnt', INT32),
+    #] + [('mc_percent{:d}'.format(i), FLOAT) for i in range(64)] + [
+    ('mc_percent', '64'+FLOAT),
+    ('ci_calib', UINT32),
+    ('calib_width', UINT32),
+    ('calib_height', UINT32),
+    ('calib_rate', UINT32),
+    ('calib_exp', UINT32),
+    ('calib_edr', UINT32),
+    ('calib_temp', UINT32),
+    #] + [('header_serial{:d}'.format(i), UINT32) for i in range(4)] + [
+    ('header_serial', '4'+UINT32),
+    ('range_code', UINT32),
+    ('range_size', UINT32),
+    ('decimation', UINT32),
+    ('master_serial', UINT32),
+    ('sensor', UINT32),
+    ('shutter_ns', UINT32),
+    ('edr_shutter_ns', UINT32),
+    ('frame_delay_ns', UINT32),
+    ('im_pos_xacq', UINT32),
+    ('im_pos_yacq', UINT32),
+    ('im_width_acq', UINT32),
+    ('im_height_acq', UINT32),
+    ('description', '4096s'),
+    ('rising_edge', BOOL),
```

Comment:
Are these attributes present in all recent versions of the CINE format? As I note in the comment below, it is not unusual to want to read older experimental movies, so compatibility should not be broken unless it is necessary.

Reply:
Yes, from version 637 up to 'description', v. 658 for 'rising_edge' (see doc p. 49).
With the _remove_trailing_x00 function, those parameters would just appear as empty string for older cine files, wouldn't they?

Reply:
Yes, please.

I'll produce some unit tests as well.

Reply:
Ok,
I wrote a new feature that stop the SETUP_FIELDS at the good location, depending on the software version with which it was recorded.
It works on your cine file, I still have to produce a small one (the ones I planned to share are too big).